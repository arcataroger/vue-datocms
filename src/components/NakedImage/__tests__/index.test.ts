import { mount } from '@vue/test-utils';
import { NakedImage } from '../';

const data = {
  alt: 'DatoCMS swag',
  aspectRatio: 1.7777777777777777,
  base64: 'data:image/jpeg;base64,<IMAGE>',
  height: 421,
  sizes: '(max-width: 750px) 100vw, 750px',
  src: 'https://www.datocms-assets.com/205/image.png?ar=16%3A9&fit=crop&w=750',
  srcSet:
    'https://www.datocms-assets.com/205/image.png?ar=16%3A9&dpr=0.25&fit=crop&w=750 187w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&dpr=0.5&fit=crop&w=750 375w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&dpr=0.75&fit=crop&w=750 562w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&dpr=1&fit=crop&w=750 750w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&dpr=1.5&fit=crop&w=750 1125w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&dpr=2&fit=crop&w=750 1500w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&dpr=3&fit=crop&w=750 2250w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&dpr=4&fit=crop&w=750 3000w',
  title: 'These are awesome, we know that.',
  width: 750,
};

const minimalData = {
  height: 421,
  src: 'https://www.datocms-assets.com/205/image.png?ar=16%3A9&fit=crop&w=750',
  width: 750,
};

const minimalDataWithRelativeUrl = {
  height: 421,
  src: '/205/image.png?ar=16%3A9&fit=crop&w=750',
  width: 750,
};

describe('Image', () => {
  describe('full data', () => {
    it('renders correctly', async () => {
      const wrapper = mount(NakedImage, {
        propsData: {
          data,
        },
      });
      await wrapper.vm.$nextTick();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('minimal data', () => {
    it('renders correctly', async () => {
      const wrapper = mount(NakedImage, {
        propsData: {
          data: minimalData,
        },
      });

      await wrapper.vm.$nextTick();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('minimalDataWithRelativeUrl', () => {
    it('renders correctly', async () => {
      const wrapper = mount(NakedImage, {
        propsData: {
          data: minimalDataWithRelativeUrl,
        },
      });

      await wrapper.vm.$nextTick();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('passing class and/or style', () => {
    it('renders correctly', async () => {
      const wrapper = mount(NakedImage, {
        propsData: {
          data: minimalData,
          pictureClass: 'picture-class-name',
          pictureStyle: { border: '1px solid red' },
          imgClass: 'img-class-name',
          imgStyle: { border: '1px solid green' },
        },
      });

      await wrapper.vm.$nextTick();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('priority=true', () => {
    it('renders correctly', async () => {
      const wrapper = mount(NakedImage, {
        propsData: {
          data: minimalData,
          priority: true,
        },
      });
      await wrapper.vm.$nextTick();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('usePlaceholder=false', () => {
    it('renders correctly', async () => {
      const wrapper = mount(NakedImage, {
        propsData: {
          data: minimalData,
          usePlaceholder: false,
        },
      });

      await wrapper.vm.$nextTick();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('explicit sizes', () => {
    it('renders correctly', async () => {
      const wrapper = mount(NakedImage, {
        propsData: {
          data: minimalData,
          sizes: '(max-width: 600px) 200px, 50vw',
        },
      });

      await wrapper.vm.$nextTick();
      expect(wrapper.html()).toMatchSnapshot();
    });
  });
});
