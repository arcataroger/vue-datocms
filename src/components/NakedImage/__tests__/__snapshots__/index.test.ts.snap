// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image explicit sizes renders correctly 1`] = `
<picture>
  <!---->
  <source srcset="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.25 187w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.5 375w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.75 562w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750 750w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=1.5 1125w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=2 1500w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=3 2250w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=4 3000w" sizes="(max-width: 600px) 200px, 50vw"><img src="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750" loading="lazy" style="width: 100%; max-width: 750px; height: auto;" referrerpolicy="no-referrer-when-downgrade">
</picture>
`;

exports[`Image full data renders correctly 1`] = `
<picture>
  <!---->
  <source srcset="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;dpr=0.25&amp;fit=crop&amp;w=750 187w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;dpr=0.5&amp;fit=crop&amp;w=750 375w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;dpr=0.75&amp;fit=crop&amp;w=750 562w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;dpr=1&amp;fit=crop&amp;w=750 750w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;dpr=1.5&amp;fit=crop&amp;w=750 1125w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;dpr=2&amp;fit=crop&amp;w=750 1500w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;dpr=3&amp;fit=crop&amp;w=750 2250w,↵https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;dpr=4&amp;fit=crop&amp;w=750 3000w" sizes="(max-width: 750px) 100vw, 750px"><img src="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750" alt="DatoCMS swag" title="These are awesome, we know that." loading="lazy" style="background-image: url(data:image/jpeg;base64,<IMAGE>); background-size: cover; background-repeat: no-repeat; background-position: 50% 50%; color: transparent; width: 100%; max-width: 750px; height: auto;" referrerpolicy="no-referrer-when-downgrade">
</picture>
`;

exports[`Image minimal data renders correctly 1`] = `
<picture>
  <!---->
  <source srcset="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.25 187w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.5 375w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.75 562w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750 750w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=1.5 1125w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=2 1500w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=3 2250w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=4 3000w"><img src="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750" loading="lazy" style="width: 100%; max-width: 750px; height: auto;" referrerpolicy="no-referrer-when-downgrade">
</picture>
`;

exports[`Image minimalDataWithRelativeUrl renders correctly 1`] = `
<picture>
  <!---->
  <source srcset="/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.25 187w,/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.5 375w,/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.75 562w,/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750 750w,/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=1.5 1125w,/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=2 1500w,/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=3 2250w,/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=4 3000w"><img src="/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750" loading="lazy" style="width: 100%; max-width: 750px; height: auto;" referrerpolicy="no-referrer-when-downgrade">
</picture>
`;

exports[`Image passing class and/or style renders correctly 1`] = `
<picture style="border: 1px solid red;" class="picture-class-name">
  <!---->
  <source srcset="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.25 187w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.5 375w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.75 562w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750 750w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=1.5 1125w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=2 1500w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=3 2250w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=4 3000w"><img src="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750" loading="lazy" style="width: 100%; max-width: 750px; height: auto; border: 1px solid green;" class="img-class-name" referrerpolicy="no-referrer-when-downgrade">
</picture>
`;

exports[`Image priority=true renders correctly 1`] = `
<picture>
  <!---->
  <source srcset="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.25 187w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.5 375w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.75 562w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750 750w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=1.5 1125w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=2 1500w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=3 2250w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=4 3000w"><img src="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750" fetchpriority="high" style="width: 100%; max-width: 750px; height: auto;" referrerpolicy="no-referrer-when-downgrade">
</picture>
`;

exports[`Image usePlaceholder=false renders correctly 1`] = `
<picture>
  <!---->
  <source srcset="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.25 187w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.5 375w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=0.75 562w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750 750w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=1.5 1125w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=2 1500w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=3 2250w,https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750&amp;dpr=4 3000w"><img src="https://www.datocms-assets.com/205/image.png?ar=16%3A9&amp;fit=crop&amp;w=750" loading="lazy" style="width: 100%; max-width: 750px; height: auto;" referrerpolicy="no-referrer-when-downgrade">
</picture>
`;
